### PR TITLE
Add Vietnamese translation of VietNam holidays

### PR DESCRIPTION
**Pull request by vunhatchuong**
## Proposed change

Add `vi` translation of Vietnam holidays.

This translation has been checked by me who was born and currently lives in Viet Nam and crossed-checked with the state [source](https://hethongphapluat.com/bo-luat-lao-dong-2019/dieu-112).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -140,3 +140,4 @@ Victor Luna
 Victor Miti
 Ville Skyttä
 Youhei Sakurai
+Vu Nhat Chuong
```
Comment (PPsyrius) [inline AUTHORS:143]:
```suggestion
Youhei Sakurai
Vu Nhat Chuong
```
Consider putting your name in the lexicographically ascending order.

File: holidays/countries/vietnam.py [hunk 2]
```diff
@@ -10,8 +10,10 @@
 #  Website: https://github.com/vacanza/python-holidays
 #  License: MIT (see LICENSE file)
 
+from gettext import gettext as tr
+
 from holidays.groups import ChineseCalendarHolidays, InternationalHolidays
-from holidays.observed_holiday_base import ObservedHolidayBase, SAT_SUN_TO_NEXT_WORKDAY
```
Comment (PPsyrius) [inline holidays/countries/vietnam.py:14]: Why the change here though?

File: holidays/countries/vietnam.py [hunk 3]
```diff
@@ -34,29 +39,29 @@ def _populate_public_holidays(self):
...
         # Liberation Day/Reunification Day
-        dts_observed.add(self._add_holiday_apr_30("Liberation Day/Reunification Day"))
+        dts_observed.add(self._add_holiday_apr_30(tr("Ngày Chiến thắng")))
 
         # International Labor Day
-        dts_observed.add(self._add_labor_day("International Labor Day"))
+        dts_observed.add(self._add_labor_day(tr("Ngày Quốc tế lao động")))
 
         # Independence Day
-        dts_observed.add(self._add_holiday_sep_2("Independence Day"))
```
Comment (PPsyrius) [inline holidays/countries/vietnam.py:59]: We might also want to take this opportunity to standardize the `en_US` localization for Vietnam holidays as well, on top of the typo fixes like `forth day` → `fourth day` from the existing VN implementation.

I'm using the [Official English translation of No. 8662/VPCP-KGVX](https://thuvienphapluat.vn/cong-van/EN/Lao-dong-Tien-luong/Official-Dispatch-8662-VPCP-KGVX-2023-Lunar-New-Year-Holiday-National-Day-Holiday-Schedules-2024/585800/tieng-anh.aspx) and [Vietnam Government News](https://en.baochinhphu.vn/viet-nam-to-have-17-public-holiday-dates-in-2024-111231225153729932.htm) as our sources for now:
```python
        # New Year's Day
        dts_observed.add(self._add_new_years_day(tr("Tết Dương lịch")))

        # Lunar New Year's Eve
        self._add_chinese_new_years_eve(tr("Giao thừa"))

        # Lunar New Year
        self._add_chinese_new_years_day(tr("Tết Nguyên Đán"))

        # Second Day of Lunar New Year
        self._add_chinese_new_years_day_two(tr("Mùng hai Tết Nguyên Đán"))

        # Third Day of Lunar New Year
        self._add_chinese_new_years_day_three(tr("Mùng ba Tết Nguyên Đán"))

        # Fourth Day of Lunar New Year
        self._add_chinese_new_years_day_four(tr("Mùng bốn Tết Nguyên Đán"))

        # Fifth Day of Lunar New Year
        self._add_chinese_new_years_day_five(tr("Mùng năm Tết Nguyên Đán"))

        # Hung Kings' Festival
        if self._year >= 2007:
            dts_observed.add(self._add_hung_kings_day(tr("Giỗ Tổ Hùng Vương")))

        # Reunification Day
        dts_observed.add(self._add_holiday_apr_30(tr("Ngày Chiến thắng")))

        # International Labor Day
        dts_observed.add(self._add_labor_day(tr("Ngày Quốc tế lao động")))

        # National Day
        dts_observed.add(self._add_holiday_sep_2(tr("Ngày Quốc khánh")))
```

**Review by PPsyrius — CHANGES_REQUESTED** [review]
Great to see new localizations :) 🇻🇳

**Review by func25 — COMMENTED** [review]
It's overall good, I have some suggestions

File: holidays/locale/en_US/LC_MESSAGES/VN.po [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+msgid "Mùng năm Tết Nguyên Đán"
+msgstr "The fifth day of Tet Holiday"
+
+msgid "Giỗ Tổ Hùng Vương"
+msgstr "Hung Kings Commemoration Day"
+
+#. Liberation Day/Reunification Day
+msgid "Ngày Chiến thắng"
+msgstr "Liberation Day/Reunification Day"
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/locale/en_US/LC_MESSAGES/VN.po:69]: The original name is the priority for us. Here it's "Ngày Chiến thắng" from [official source](https://hethongphapluat.com/bo-luat-lao-dong-2019/dieu-112).

File: holidays/countries/vietnam.py [hunk 5]
```diff
@@ -34,29 +39,38 @@ def _populate_public_holidays(self):
...
-        dts_observed.add(self._add_holiday_apr_30("Liberation Day/Reunification Day"))
+        dts_observed.add(self._add_holiday_apr_30(tr("Ngày Chiến thắng")))
 
         # International Labor Day
-        dts_observed.add(self._add_labor_day("International Labor Day"))
+        dts_observed.add(self._add_labor_day(tr("Ngày Quốc tế Lao động")))
 
         # Independence Day
-        dts_observed.add(self._add_holiday_sep_2("Independence Day"))
+        dts_observed.add(self._add_holiday_sep_2(tr("Ngày Quốc khánh Việt Nam")))
```
Comment (KJhellico) [inline holidays/countries/vietnam.py:73]:
```suggestion
        dts_observed.add(self._add_holiday_sep_2(tr("Quốc khánh")))
```
As is source. And with en_US translation "National Day" ([source](https://vbpl.vn/TW/Pages/vbpqen-toanvan.aspx?ItemID=11135)).

File: holidays/locale/en_US/LC_MESSAGES/VN.po [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+"Generated-By: Lingva 5.0.4\n"
+
+#. %s (observed).
+#, c-format
+msgid "%s (nghỉ bù)"
+msgstr "%s (observed)"
+
+#. New Year's Day
+msgid "Tết Dương lịch"
+msgstr "International New Year's Day"
```
Comment (KJhellico) [inline holidays/locale/en_US/LC_MESSAGES/VN.po:35]:
```suggestion
msgstr "New Year's Day"
```

File: holidays/countries/vietnam.py [hunk 7]
```diff
@@ -34,29 +39,38 @@ def _populate_public_holidays(self):
...
+        # Fourth Day of Lunar New Year
+        self._add_chinese_new_years_day_four(tr("Mùng bốn Tết Nguyên Đán"))
+
+        # Fifth Day of Lunar New Year
+        self._add_chinese_new_years_day_five(tr("Mùng năm Tết Nguyên Đán"))
+
         if self._year >= 2007:
-            dts_observed.add(self._add_hung_kings_day("Hung Kings Commemoration Day"))
+            # Hung Kings' Festival
+            dts_observed.add(self._add_hung_kings_day(tr("Giỗ Tổ Hùng Vương")))
```
Comment (KJhellico) [inline holidays/countries/vietnam.py:64]:
```suggestion
            # Hung Kings' Commemoration Day
            dts_observed.add(self._add_hung_kings_day(tr("Ngày Giỗ Tổ Hùng Vương")))
```
To follow [official source](https://hethongphapluat.com/bo-luat-lao-dong-2019/dieu-112).

File: holidays/locale/en_US/LC_MESSAGES/VN.po [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+msgid "Mùng bốn Tết Nguyên Đán"
+msgstr "The fourth day of Tet Holiday"
+
+#. Fifth Day of Lunar New Year
+msgid "Mùng năm Tết Nguyên Đán"
+msgstr "The fifth day of Tet Holiday"
+
+#. Hung Kings' Festival
+msgid "Giỗ Tổ Hùng Vương"
+msgstr "Hung Kings Commemoration Day"
```
Comment (KJhellico) [inline holidays/locale/en_US/LC_MESSAGES/VN.po:63]:
```suggestion
#. Hung Kings' Commemoration Day
msgid "Giỗ Tổ Hùng Vương"
msgstr "Hung Kings' Commemoration Day"
```

File: holidays/locale/en_US/LC_MESSAGES/VN.po [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+msgid "Ngày Chiến thắng"
+msgstr "Liberation Day/Reunification Day"
+
+#. International Labor Day
+msgid "Ngày Quốc tế Lao động"
+msgstr "International Labor Day"
+
+#. Independence Day
+msgid "Ngày Quốc khánh Việt Nam"
+msgstr "Vietnam's Independence Day"
```
Comment (KJhellico) [inline holidays/locale/en_US/LC_MESSAGES/VN.po:75]:
```suggestion
#. National Day
msgid "Quốc khánh"
msgstr "National Day"
```

**Review by KJhellico — COMMENTED** [review]
@vunhatchuong, great work! Look at some naming suggestions, please.

**Comment by KJhellico** [comment]
@vunhatchuong,
I have a few more questions about the names of holidays in Vietnam. I hope you can tell me something.

1. How official (and common) are the names “Second Day of Lunar New Year”, “Third Day of Lunar New Year”, etc.? Labor Code simply says “Tết Âm lịch: 05 ngày” (“Lunar new year festival: 5 days”). Reputable unofficial holiday portals (Timeanddate, OfficeHolidays) only list New Year's Eve / New Year / Tet holiday (for the rest of the days).

2. What is the Vietnamese name for a day off moved from a weekend (for example, April 29, 2024)? In Python Holidays, we call such days "substituted holidays". I would like to add such holidays for Vietnam, but I can't find the original name for them. In fact, I need the correct translation of sentence “Day off (substituted from 4/5/2024)”.

I would be thankful for your help!

**Comment by vunhatchuong** [comment]
@KJhellico,
Sorry for the late response, in this PR I mainly tackle translation since I'm not confident about how Tet is calculated and I can't find any official sources as proof.

1. “Second Day of Lunar New Year”, ... is a common way to call it. I can't find any source to prove if the State calls them as either.
Tet holiday technically only lasts 3 days or 7 (I don't know) starting from 1/1 on the Lunar calendar but the Labor Code allows people to have a **minimum** of 5 days off hence “Lunar New Year: 5".

3.  Isn't "substituted holiday" the same as "observed holiday"? This is why I translate `"% (observed)"` as `"% (nghỉ bù)"`, "nghỉ bù" means "compensatory leave".

---

Regarding days off because of a holiday, the Tet holiday calculation is also incorrect because it starts from the 29th in the Lunar calendar + another 4 days (**minimum** of 5 days). So in 2024, people were already off work on 2024-02-08 and not 2024-02-09.

So it doesn't matter if Feb has 29 or 30 days, people will always leave work starting from the 29th.

**Comment by KJhellico** [comment]
@vunhatchuong, thank you for your answer!

> 1. “Second Day of Lunar New Year”, ... is a common way to call it. I can't find any source to prove if the State calls them as either.
>    Tet holiday technically only lasts 3 days or 7 (I'm unsure) starting from 1/1 on the Lunar calendar but the Labor Code allows people to have a **minimum** of 5 days off starting on Dec 29th hence “Lunar New Year: 5".

OK, I get it. And what are the extra days called (as Feb 8 and Feb 14 in 2024, Jan 20 and Jan 26 in 2023, ...)?

> 2. Isn't "substituted holiday" the same as "observed holiday"? This is why I translate `"% (observed)"` as `"% (nghỉ bù)"`, "nghỉ bù" means "compensatory leave".

In Python holidays we call it "observed holiday" when a holiday falls on weekend and there is an additional non-working day. For example, in 2023, New Year's Day (Jan 1) fell on a Sunday, and Jan 2, Monday, was a "observed" holiday.
"Substituted holiday" - it's when working day is transfer to weekend (to make a long holiday). For example, in 2024 working day from Apr 29 (Mon) was moved to May 4 (Sun), and there were no working days from Apr 27 (Sat) to May 1 (Wed, International Labor Day).

So your current translation is correct. And I'm looking for the translation for name of such days as Apr 29, 2024.

> Regarding days off because of a holiday, the Tet holiday calculation is currently incorrect because it starts from the 29th in the Lunar calendar + another 4 days (**minimum** of 5 days). So in 2024, people were already off work on 2024-02-08 and not 2024-02-09.

Yes, I'm already working on it. I think I understand this scheme.

**Comment by vunhatchuong** [comment]
@KJhellico 
> OK, I get it. And what are the extra days called (as Feb 8 and Feb 14 in 2024, Jan 20 and Jan 26 in 2023, ...)?

They just call it "The X Day of Lunar New Year". "ngày" and "mùng" or "mồng" just mean "day" but "mùng" is used to refer to the Lunar day. So "ngày 1" means "day 1 on the Gregorian Calendar" and "mùng 1" means "day 1 on the Lunar Calendar".

> And I'm looking for the translation for name of such days as Apr 29, 2024

I can't find any specific name for it, so the direct translation is "nghỉ thay thế" which sounds good enough.

**Comment by KJhellico** [comment]
> They just call it "The X Day of Lunar New Year".

But how about Lunar New Year's Eve and the day before it? As Feb 9 and Feb 8 in 2024?

> I can't find any specific name for it, so the direct translation is "nghỉ thay thế" which sounds good enough.

For such holidays, we also mention the day on from which the day off is moved (“Day off (substituted from 4/5/2024)” in English). Google Translate gave me "Ngày nghỉ (thay thế từ ngày 4/5/2024)". How correct is this?

**Comment by vunhatchuong** [comment]
> But how about Lunar New Year's Eve and the day before it? As Feb 9 and Feb 8 in 2024?

There are no special names for it. You can see [here](https://thuvienphapluat.vn/phap-luat/ho-tro-phap-luat/tet-2024-vao-ngay-nao-duong-lich-lich-nghi-tet-nguyen-dan-2024-va-duong-lich-2024-nhu-the-nao-179705-126634.html) and other sources they call it "29 Tet" and 30 Tet". But my current translate for Lunar New Year's Eve is "Giao thừa" is correct.

> "Ngày nghỉ (thay thế từ ngày 4/5/2024)". How correct is this?

I'd say "Ngày nghỉ (thay cho ngày 4/5/2024)" is more correct.

**Comment by KJhellico** [comment]
> There are no special names for it. You can see [here](https://thuvienphapluat.vn/phap-luat/ho-tro-phap-luat/tet-2024-vao-ngay-nao-duong-lich-lich-nghi-tet-nguyen-dan-2024-va-duong-lich-2024-nhu-the-nao-179705-126634.html) and other sources they call it "29 Tet" and 30 Tet". 

Yes, I understood. But we need to name them somehow, with the possibility of translation into other languages...

> But my current translate for Lunar New Year's Eve is "Giao thừa" is correct.

Google translates it just as "New Year's Eve".  What would the Vietnamese name be for the Dec 31 holiday ("New Year's Eve") that exists in some countries?

> I'd say "Ngày nghỉ (thay cho ngày 4/5/2024)" is more correct.

Thank you! 👍

**Comment by vunhatchuong** [comment]
> Yes, I understood. But we need to name them somehow, with the possibility of translation into other languages...

Yeah this is why I'm not confident... but If it was up to me, I will name it "29 Tết" and "30 Tết".

> Google translates it just as "New Year's Eve". What would the Vietnamese name be for the Dec 31 holiday ("New Year's Eve") that exists in some countries?

I guess to be precise it's "Lunar New Year's Eve", the reason I shortened it is that Dec 31 "New Year's Eve" is not important in Vietnam so it's not named.

**Comment by KJhellico** [comment]
> I guess to be precise it's "Lunar New Year's Eve", the reason I shortened it is that Dec 31 "New Year's Eve" is not important in Vietnam so it's not named.

Maybe "Giao thừa Tết Nguyên Đán" would be better, In terms of generality?

**Review by KJhellico — DISMISSED** [review]
LGTM! 👍

**Review by PPsyrius — APPROVED** [review]
LGTM 👍🇻🇳

**Review by arkid15r — COMMENTED** [review]
@vunhatchuong thank you for the great contribution and `VN` holidays localization!

@func25 @KJhellico @PPsyrius thanks for reviewing this!

**Review by arkid15r — APPROVED** [review]

**Pull request merged**